docs/gallery/general/plot_configurator.py

```diff
@@ -0,0 +1,108 @@
+"""
+TypeConfigurator
+=======
+
+This is a user guide for how to take advantage of the
+:py:class:`~hdmf.term_set.TypeConfigurator` class by creating configuration files in
+order to more easily validate terms within datasets or attributes.
+
+Introduction
+-------------
+Users do not directly interact with the :py:class:`~hdmf.term_set.TypeConfigurator` class.
+Instead, users will create a configuration YAML file that outlines the fields (within a neurodata type)
+they want to be validated. After creating the configuration file, users will need to load the
+configuration with the :py:func:`~pynwb.load_type_config` method.
```

Comment:
A user should not need to know about the TypeConfigurator class, so I think we do not need to talk about it. It's basically an internal class, like the TypeMap, BuildManager, etc.